[알린] Store step6 - PR

jinios opened the pull request:
이미지 병렬처리 Step6 PR보냅니다! :)

jinios committed:
Feat: Add Item Image download feature #10

jinios committed:
Feat: Add deleteCache() for DEBUG mode #10

jinios committed:
Feat: Show itemImage on cell #9, #10
- Add ImageSetter class for downloading and saving images in cache directory

jinios committed:
Docs: Update README - step6

godrm commented on StoreApp/StoreApp/Model/ImageSetter.swift:
```diff
@@ -0,0 +1,44 @@
+                        } else {
+                            print("MOVE Error!")
+                        }
+                    } catch { print("MOVE Error!") }
```
여기 에러들은 로그 찍는거 말고 어떻게 할 수 있을까요? 
네트워크는 항상 실패할 수 있기 때문에 대비를 해야합니다.

jinios replied:
네트워크 에러인 경우에는 사용자에게 알려주고 refresh를 하거나 나중에 다시 시도해달라고 해야할 것 같습니다. 하지만 네트워크가 문제라면 refresh를 하더라도 이미지를 표시할 수는 없을것같습니다....! 
(모든 이미지를 앱에 저장해놓지 않는 이상 다른 방법이 있을까요? 🤔)
하지만 적어도 사용자에게 문제가 있는걸 알려줘야한다고 생각합니다. 수정해보겠습니다! :)

godrm replied:
맞아요 WiFi가 안되거나 여러 이미지가 실패하면 사용자에게 못 받았다고 알림창을 여러개 띄우는걸 말하는건 아니겠죠?
네트워크에 접근하는 코드는 항상 실패를 확인해서 상위 모듈에 전달하는 구조가 있어야 해요.
그게 return type 이든지 throw exception 이든지 
레벨2에서 exception을 메인으로 전달하던 기억을 떠올려보세요 :)

jinios committed:
Feat: Show refresh Image when error occurred #11
- When error occurred in network, or in reading file in cache directory, refresh Image is shown on the cell

jinios committed:
Docs: Update README - step6-2

jinios commented:
이미지 로드에 문제가 있으면 셀에 리프레시이미지를 띄우도록 수정했습니다!
